cdap-securestore-ext-cloudkms/src/main/java/co/cask/cdap/securestore/gcp/cloudkms/SecretInfo.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.securestore.gcp.cloudkms;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class used to serialize and deserialize the secret and associated metadata. Used by {@link SecretEncoderDecoder}.
+ */
+public final class SecretInfo {
```

Comment:
Is this class different from Secret?

Is the intention that implementation will have their own definition?

Reply:
It contains the same information. It was added to address: https://github.com/cdapio/cdap/pull/10944#discussion_r247200130